### PR TITLE
feat: add CLI tool to enable the complete creation of core components…

### PR DESCRIPTION
**Pull request by 0xff-dev**
… based on specified parameters

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: add CLI tool to enable the complete creation of core components…

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer

```shell
./main install --namespace=u4a-system --component-store=false --cc-version=0.1.0 --u4a-version=0.1.7 --node-name=kubebb-core-control-plane
```

### CODE REVIEW

File: pkg/cmd/config.go [hunk 1]
```diff
@@ -17,102 +17,87 @@ limitations under the License.
...
+		URL:  KUBEBBOFFICIALREPOADDRESS,
 	}
-
-	cmd.Flags().BoolVar(&installClusterComponent, "cluster-component", false, "install cluster-component?")
-	cmd.Flags().BoolVar(&installU4A, "u4a", false, "install u4a?")
-	cmd.Flags().BoolVar(&installComponentStore, "component-store", false, "install component-store?")
-	return cmd
+	logger := logr.Logger{}
+	ctx := context.Background()
+	if err := helm.RepoRemove(ctx, logger, entry.Name); err != nil && !(err.Error() == "no repositories configured" || err.Error() == "no repo named \"kubebb-official-repo\" found") {
```
Comment (Abirdcfly) [inline pkg/cmd/config.go:97]: kubebb-official-repo -> KUBEBBOFFICIALREPO

File: pkg/cmd/config.go [hunk 2]
```diff
@@ -17,102 +17,87 @@ limitations under the License.
...
+		fmt.Fprintf(os.Stderr, "failed to add repo %s", err)
+		return
+	}
+	fmt.Println("try to add repo")
+	_ = helm.RepoAdd(ctx, logger, entry, 5*time.Second)
```
Comment (Abirdcfly) [inline pkg/cmd/config.go:102]: do we need a repo update?

File: pkg/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		u4aExtraConfString            string
+		componentStroeExtraConfString string
+		kubeconfig                    string
+		nodeName                      string
+	)
+	cmd := &cobra.Command{
+		Use:  "install",
+		Long: "install core, cluster-component",
+		Run: func(cmd *cobra.Command, args []string) {
+			background := context.Background()
```
Comment (Abirdcfly) [inline pkg/cmd/root.go:52]:
```suggestion
			background := cmd.Context()
```

File: pkg/cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+				taskConf.NodeIP = nodeIP
+
+				if !taskConf.Install && !taskConf.Upgrade {
+					fmt.Printf("skip %s", taskConf.RegisterName)
+					continue
+				}
+				instance := GetInstaller(taskConf.RegisterName)
+				if taskConf.Install && instance != nil {
+					installer := instance(&taskConf)
+					fmt.Printf("[%d] install %s", step, taskConf.RegisterName)
```
Comment (Abirdcfly) [inline pkg/cmd/root.go:111]: look like this step is same with idx ?

**Review by Abirdcfly — APPROVED** [review]

File: pkg/cmd/common_componentplan.go [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  enabled: false
+  storageClass: openebs-hostpath
+#  ldapImage: 172.22.96.19/u4a_system/openldap:1.5.0
+#  ldapOrg: test
+#  ldapDomain: test.com
+#  ldapAdminPwd: xxx
+#  ldapAdminImage: 172.22.96.19/u4a_system/phpldapadmin:stable
+
+cluster-component:
+  enabled: false`
```
Comment (nkwangleiGIT) [inline pkg/cmd/common_componentplan.go:224]: yaml 具体配置最好别写在代码里吧，可以放到某个配置文件，然后这里去 read

File: pkg/cmd/common_componentplan.go [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+	Enroll(COMPONENTSTORE, CommonInstaller)
+}
+
+func initCli() {
+	cfg, err := ctrl.GetConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("init controller runtime client")
```
Comment (nkwangleiGIT) [inline pkg/cmd/common_componentplan.go:239]: 如果需要记录日志，换成 logger？

File: pkg/cmd/core.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+		fmt.Fprintf(os.Stderr, "can't get cluster config %s", err)
+		return fmt.Errorf("[%s] Failed to get cluster config %w", CORE, err)
+	}
+	getter := genericclioptions.ConfigFlags{
+		APIServer:   &cfg.Host,
+		CAFile:      &cfg.CAFile,
+		BearerToken: &cfg.BearerToken,
+		Namespace:   &c.cfg.Namespace,
+	}
+	fmt.Printf("\t[%s] construct new helm wrapper\n", CORE)
```
Comment (nkwangleiGIT) [inline pkg/cmd/core.go:60]: 换成 logger？

File: pkg/cmd/core.go [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+	logger := logr.Logger{}
+	hl, err := helm.NewHelmWrapper(&getter, c.cfg.Namespace, logger)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "new helm wrapper error %s", err)
+		return err
+	}
+
+	actionClient := hl.GetDefaultInstallCfg()
+	actionClient.Version = c.cfg.Version
+	fmt.Printf("\t[%s] helm install %s %s/kubebb-core\n", CORE, CORE, KUBEBBOFFICIALREPO)
```
Comment (nkwangleiGIT) [inline pkg/cmd/core.go:70]: 换成 logger？

File: pkg/cmd/common_componentplan.go (hunk 5 above)
Comment (0xff-dev) [reply pkg/cmd/common_componentplan.go:224]: 目前考虑到u4a跟随core发版的版本已经稳定，不会出现大升级以及需要快速实现cli，快速使用，如果将values.yaml放到配置文件里，增加了用户寻找配置路径的步骤。理想情况是直接通过 `corctl install `完成core，cluster-component，u4a，componet-stroe这些组件。所以这里先把配置文件写到代码里了。

File: pkg/cmd/common_componentplan.go (hunk 6 above)
Comment (bjwswang) [reply pkg/cmd/common_componentplan.go:239]: +1

**Review by bjwswang — APPROVED** [review]

**Comment by bjwswang** [comment]
@0xff-dev we should have a step-by-step guidence to let user configure the must-required fileds like

```shell
cli init
> Input the namespace for kubebb: xxx
> Select a node for ingress:   xxx
```

**Pull request merged**